nailgun/entities.py

```diff
@@ -2740,17 +2870,41 @@ def read(self, entity=None, attrs=None, ignore=()):
         return super(PuppetModule, self).read(entity, attrs, ignore)
 
 
-class Realm(Entity):
+class Realm(
```

Comment:
I can create a Realm with the current code. There's a catch, unfortunately - the `create` method is broken. See the example code below.

The fragility of NailGun with regards to the names that the server may return is very annoying. I've described the issue in #110 and #114. I've already implemented a fix, too, in my "robustness" branch.

```python
>>> import robottelo; from nailgun import entities
>>> smart_proxy = entities.SmartProxy(id=1).read()
>>> realm = entities.Realm(
...     realm_proxy=smart_proxy,
...     realm_type='Active Directory',
...     name='foo',
... ).create(create_missing=False)
Traceback (most recent call last):
  File "<stdin>", line 5, in <module>
  File "/home/ichimonji10/code/nailgun/nailgun/entity_mixins.py", line 766, in create
    return self.read(attrs=self.create_json(create_missing))
  File "/home/ichimonji10/code/nailgun/nailgun/entities.py", line 2909, in read
    return super(Realm, self).read(entity, attrs, ignore)
  File "/home/ichimonji10/code/nailgun/nailgun/entity_mixins.py", line 621, in read
    in attrs[field_name + 's']  # e.g. "users"
KeyError: 'organizations'
>>> attrs = entities.Realm(
...     realm_proxy=smart_proxy,
...     realm_type='Active Directory',
...     name='foo2',
... ).create_json(create_missing=False)
>>> attrs['id']
2
>>> realm = entities.Realm(id=2).read()
>>> realm.id, realm.name
(2, u'foo2')
```